src/middlewares/__tests__/isAuth.test.js

```diff
@@ -0,0 +1,141 @@
+const jwt = require('jsonwebtoken');
+const isAuth = require('../isAuth');
+
+const validToken = { tokenValue: 'validToken', expires: 'validTime' };
+const validTokenWithNoValidTime = {
+  tokenValue: 'invalidToken',
+  expires: 'validTime',
+};
+const invalidToken = {
+  tokenValue: 'invalidToken',
+  expires: 'validTime',
+};
+const invalidTokenWithInvalidTime = {
+  tokenValue: 'invalidToken',
+  expires: 'invalidTime',
+};
+const user = {
+  id: 1,
+  username: 'name',
+  email: 'someemail@gmail.com',
+};
+
+const res = {
+  status: jest.fn().mockReturnThis(),
+  json: jest.fn().mockReturnThis(),
+  clearCookie: jest.fn().mockReturnThis(),
+};
+const next = jest.fn().mockReturnThis();
```

Comment:
You should add `mockReturnThis()` only for functions that will be chained in your implementation.